src/workflows/lulc/generate_change.py

```diff
@@ -0,0 +1,158 @@
+from src.local_stac.generate import generate_stac, prepare_stac_item
+from src.raster_utils.build import build_raster_array
+from src.raster_utils.save import save_cog
+from src.utils.logging import get_logger
+
+if TYPE_CHECKING:
+    import xarray
+    from pystac import Item
+
+_logger = get_logger(__name__)
+
+
+@click.command(help="Generate LULC change")
+@click.option("--aoi", required=True, help="The area of interest as GeoJSON in EPSG:4326")
+@click.option("--start_date", required=True, help="Start date in ISO 8601 used to search input data")
+@click.option("--end_date", required=True, help="End date in ISO 8601 used to search input data")
+def generate_lulc_change(aoi: str, start_date: str, end_date: str) -> None:
+    # Connect to STAC API
+    catalog = Client.open(consts.stac.CEDA_CATALOG_API_ENDPOINT)
+    stac_collection = consts.stac.CEDA_ESACCI_LC_COLLECTION_NAME
+
+    _logger.info(
+        "Running with:\n%s", json.dumps({"aoi": aoi, "start_date": start_date, "end_date": end_date}, indent=4)
+    )
+
+    # Transferring the AOI
+    aoi_geojson = json.loads(aoi)
+    if aoi_geojson["type"] != "Polygon":
+        msg = "Provided GeoJSON is not a polygon"
+        raise ValueError(msg)
+    aoi_polygon = shape(aoi_geojson)
+
+    # Querying the data
+    search = catalog.search(
+        filter_lang="cql2-json",
+        filter={
+            "op": "and",
+            "args": [
+                {"op": "s_intersects", "args": [{"property": "geometry"}, mapping(aoi_polygon)]},
+                {"op": "=", "args": [{"property": "collection"}, stac_collection]},
+                {"op": ">=", "args": [{"property": "datetime"}, start_date]},
+                {"op": "<=", "args": [{"property": "datetime"}, end_date]},
+            ],
+        },
+    )
+    items = sorted(search.items(), key=lambda item: item.datetime)
+
+    # Calculating lulc change
+    classes_shares: list[dict[str, int]] = []
+    stac_items: list[Item] = []
+
+    progress_bar = tqdm(items, desc="Processing items")
+    for idx, item in enumerate(progress_bar):
+        progress_bar.set_description(f"Working with: {item.id}")
+
+        # Get additional properties of the item
+        epsg = int(item.properties.get("proj:code").replace("EPSG:", ""))
+        transform = item.properties.get("proj:transform")
+
+        # Build array
+        raster_arr = build_raster_array(
+            item=item,
+            bbox=aoi_polygon.bounds,
+            epsg=epsg,
+            assets=["GeoTIFF"],
+            resolution=(
+                float(item.properties.get("geospatial_lon_resolution")),
+                float(item.properties.get("geospatial_lat_resolution")),
+            ),
+        )
+
+        # Count occurrences for each class
+        classes_shares.append(_get_counts_for_classes(raster_arr))
+
+        # Save COG with lulc change values in metadata
+        # Change is calculated based on the previously processed item
+        if idx > 0:
```

Comment:
As discussed, let's change this logic:
- Instead of calculating diff between 2 consecutive items
- Calculate area per item (km sq)
- Calculate % of  given class for each item
- Always include all classes even if it's not present in the image